src/context/LocationsContext.js

```diff
@@ -29,23 +30,34 @@ const LocationsContextProvider = (props) => {
 
     if (navigator.geolocation && allowLocation) {
       setLoading({ loading: true, message: 'loading' });
-      navigator.geolocation.getCurrentPosition(
-        (position) => {
-          setUserLocation(position.coords);
-          setLoading({ loading: false, message: '' });
-        },
-        (error) => {
-          console.error(error);
-          setLoading({
-            loading: false,
-            message: 'location services turned off',
-          });
-        },
+      setId((id) =>
+        navigator.geolocation.watchPosition(
+          (position) => {
+            setUserLocation(position.coords);
+            setLoading({ loading: false, message: '' });
+          },
+          (error) => {
+            console.error(error);
+            setLoading({
+              loading: false,
+              message: 'location services turned off',
+            });
+          },
+        ),
+      );
+    }
+    if (navigator.geolocation && !allowLocation) {
+      navigator.geolocation.clearWatch(id);
+      setUserLocation(undefined);
+      getLocations(
+        defaultCoordinates.lat,
```

Comment:
Ah, OK. I looked at the ticket again and I see that I said “return to the default state”, which is probably why y’all used the default coordinates here. What I think would be a better experience is just leaving the map centered wherever it currently is. I can’t believe you didn’t read my mind! 😅 

I’ll think of a better way to phrase that for future teams!

Reply:
haha all good, I removed this so now the map is left in its current view and the user location icon is toggled off